src/Symfony/Bridge/Monolog/Processor/TokenProcessor.php

```diff
@@ -30,7 +30,7 @@ public function __construct(TokenStorageInterface $tokenStorage)
     public function __invoke(array $records)
     {
         $records['extra']['token'] = null;
-        if (null !== $token = $this->tokenStorage->getToken()) {
+        if (null !== $token = $this->tokenStorage->getToken(false)) {
```

Comment:
I have the feeling this is a step backwards in terms of code quality :cry: 

When I see this and then look at the interface I'm confused:

https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Core/Authentication/Token/Storage/TokenStorageInterface.php#L28

So here we kind of code against the implementation `UsageTrackingTokenStorage` and not against the abstraction/interface anymore.

Reply:
Also if someone decorated the original `TokenStorage` service this will not work properly, right? (as the argument will be missing in any `parent::getToken()` call because its simply not part of the contract).

Reply:
> we kind of code against the implementation UsageTrackingTokenStorage

fixed by adding a new `UsageTrackingTokenStorageInterface`

> if someone decorated the original security.token_storage service this will not work properly, right?

The behavior will be the same as currently. That's the typical downside with service decoration: you're required to adapt to changes done at the decorated level. It's not very different from being required to write new code to leverage any new features. There is no way around.

Reply:
the issue here is that your new interface adds a new argument to the *same* method, making it really weird for decorators

Reply:
I don't understand what is weird. This works fine.

Reply:
@stof I think that's OK. We don't want ppl to have to use two different methods for fetching the token. And this is in a separate interface (`UsageTrackingTokenStorageInterface`) so that's documented. We don't want ppl to decorate blindly btw.